cdap-app-fabric/src/main/java/io/cdap/cdap/config/PreferencesService.java

```diff
@@ -101,10 +115,9 @@ private void setConfig(EntityId entityId,
   /**
    * Validate the profile status is enabled and set the preferences
    */
-  private void setConfig(ProfileStore profileStore, PreferencesTable preferencesTable, EntityId entityId,
-                         Map<String, String> propertyMap)
-    throws NotFoundException, ProfileConflictException, BadRequestException, IOException {
-
+  private void setConfig(ProfileStore profileStore, PreferencesTable preferencesTable,
+                         @Name("entityId") EntityId entityId, Map<String, String> propertyMap) throws Exception {
+    authorizationEnforcer.enforce(entityId, authenticationContext.getPrincipal(), Action.EXECUTE);
     boolean isInstanceLevel = entityId.getEntityType().equals(EntityType.INSTANCE);
     NamespaceId namespaceId = isInstanceLevel ?
       NamespaceId.SYSTEM : ((NamespacedEntityId) entityId).getNamespaceId();
```

Comment:
It looks like there is a distinction which always attempts to enforce on a namespace entity. Should we performing the enforcement at the lower level in this case?

Reply:
As discussed, auth extension can chose to ignore if it is an instance id